app/controllers/users_controller.rb

```diff
@@ -1,5 +1,13 @@
 class UsersController < ApplicationController
-  def show
-    @user = User.find_by!(username: params.fetch(:username))
-  end
+  before_action :set_user, only: %i[ show liked feed followers following discover ]
+
+  private
+
+    def set_user
+      if params[:username]
+        @user = User.find_by!(username: params.fetch(:username))
+      else
+        @user = current_user
+      end
+    end
```

Comment:
While I understand what this code is doing, I don't understand why you need the first portion, i.e., that in the "if". Why would you need to define @user using parameters versus always using 'current_user'? My instinct is that you may use a method that requires the current user to send another user's username as a parameter and this would set @user to be that username (if it exists). Could you provide a scenario on Photogram Industrial where the if would be true?

Reply:
This conditional is required because we're reusing the same action, `Users#feed`, for two routes:

 - `root "users#feed"`
 - `get "/:username/feed" => "users#feed"`

When someone visits the root path, their username is not in the URL, so we have no `params[:username]` to work with, so we fall back using to the current user.